src/KubernetesClient/TimeoutHandler.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.WebUtilities;
+
+namespace informers
+{
+    public class TimeoutHandler : DelegatingHandler
```

Comment:
Seems like it could use a better class name, since it sounds like a generic timeout handler but it's watch-specific. Also, is the namespace ("informers") correct?

Reply:
I took a second look and realized all this logic belongs in the existing `WatcherDelegatingHandler`. Code has been updated